tests/test_sagemaker/test_sagemaker_endpoint.py

```diff
@@ -0,0 +1,247 @@
+        )
+    assert_equal(e.exception.response["Error"]["Code"], "ValidationException")
+    expected_message = "Value '{}' at 'instanceType' failed to satisfy constraint: Member must satisfy enum value set: [".format(
+        instance_type
+    )
+    assert_true(expected_message in e.exception.response["Error"]["Message"])
+
+
+@mock_sagemaker
+def test_create_endpoint():
+    sagemaker = boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+    endpoint_name = "MyEndpoint"
+    with assert_raises(ClientError) as e:
+        sagemaker.create_endpoint(
+            EndpointName=endpoint_name, EndpointConfigName="NonexistentEndpointConfig"
+        )
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith(
+            "Could not find endpoint configuration"
+        )
+    )
+
+    model_name = "MyModel"
+    _create_model(sagemaker, model_name)
+
+    endpoint_config_name = "MyEndpointConfig"
+    _create_endpoint_config(sagemaker, endpoint_config_name, model_name)
+
+    resp = sagemaker.create_endpoint(
+        EndpointName=endpoint_name,
+        EndpointConfigName=endpoint_config_name,
+        Tags=GENERIC_TAGS_PARAM,
+    )
+    resp["EndpointArn"].should.match(
+        r"^arn:aws:sagemaker:.*:.*:endpoint/{}$".format(endpoint_name)
+    )
+
+    resp = sagemaker.describe_endpoint(EndpointName=endpoint_name)
+    resp["EndpointArn"].should.match(
+        r"^arn:aws:sagemaker:.*:.*:endpoint/{}$".format(endpoint_name)
+    )
+    resp["EndpointName"].should.equal(endpoint_name)
+    resp["EndpointConfigName"].should.equal(endpoint_config_name)
+    resp["EndpointStatus"].should.equal("InService")
+    assert_true(isinstance(resp["CreationTime"], datetime.datetime))
+    assert_true(isinstance(resp["LastModifiedTime"], datetime.datetime))
+    resp["ProductionVariants"][0]["VariantName"].should.equal("MyProductionVariant")
+
+    resp = sagemaker.list_tags(ResourceArn=resp["EndpointArn"])
+    assert_equal(resp["Tags"], GENERIC_TAGS_PARAM)
+
+
+@mock_sagemaker
+def test_delete_endpoint():
+    sagemaker = boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+    model_name = "MyModel"
+    _create_model(sagemaker, model_name)
+
+    endpoint_config_name = "MyEndpointConfig"
+    _create_endpoint_config(sagemaker, endpoint_config_name, model_name)
+
+    endpoint_name = "MyEndpoint"
+    _create_endpoint(sagemaker, endpoint_name, endpoint_config_name)
+
+    sagemaker.delete_endpoint(EndpointName=endpoint_name)
+    with assert_raises(ClientError) as e:
+        sagemaker.describe_endpoint(EndpointName=endpoint_name)
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith("Could not find endpoint")
+    )
+
+    with assert_raises(ClientError) as e:
+        sagemaker.delete_endpoint(EndpointName=endpoint_name)
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith("Could not find endpoint")
+    )
+    pass
```

Comment:
Don't need the pass-statements :)

Reply:
Me and my extra debugging breakpoint lines!